### PR TITLE
Add JavaDoc for TypeReferenceLocation enum values

### PR DESCRIPTION
**Pull request by sgilda**

### CODE REVIEW

**Comment by sgilda** [comment]
Feel free to reword as needed.

**Comment by windupgithubbot1** [comment]
Can one of the admins verify this patch?

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/ast/TypeReferenceLocation.java [hunk 1]
```diff
@@ -7,21 +7,69 @@
  */
 public enum TypeReferenceLocation
 {
+    /**
+     *  A Java class imports the type reference.
```
Comment (jsight) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/ast/TypeReferenceLocation.java]: I would drop the word "reference" on this..

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/ast/TypeReferenceLocation.java [hunk 2]
```diff
@@ -7,21 +7,69 @@
...
+     */
     IMPORT("Import of"),
+    /**
+     * A Java class declares the type reference.
```
Comment (jsight) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/ast/TypeReferenceLocation.java]: Same as above.

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/ast/TypeReferenceLocation.java [hunk 3]
```diff
@@ -7,21 +7,69 @@
...
+    /**
+     * A Java class calls the referenced method.
+     */
     METHOD_CALL("Calls method"),
+    /**
+     * A Java class declares the referenced method parameter.
+     */
     METHOD_PARAMETER("Method parameter"),
+    /**
+     * A Java class declares the referenced annotation.
```
Comment (jsight) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/ast/TypeReferenceLocation.java]: It doesn't declare the annotation, it references the annotation.

**Comment by lincolnthree** [comment]
Thank you, Sande! We can still improve this, but this is a good start.

**Pull request merged**